### PR TITLE
Issue #1376 Improve detection of Hyper-V and error msg when VirtualBox is used as the Hypervisor

### LINKED ISSUE #1376: Improve error message on start for using Virtualbox when Hyper-V is enabled
While trying to start with minishift with Virtualbox driver when Hyper-V is enabled, I am getting error messages which looks quite ugly and not user friendly.

```
PS C:\Users\budhram> minishift start --vm-driver virtualbox --iso-url file://C:/ms_cache/iso/1.1.0/minishift-b2d.iso
-- Starting local OpenShift cluster using 'virtualbox' hypervisor ...
-- Minishift VM will be configured with ...
   Memory:    2 GB
   vCPUs :    2
   Disk size: 20 GB
-- Starting Minishift VM ..... FAIL E0913 12:36:13.753837   10688 start.go:339] Error starting the VM: Error creating the VM. Error with pre-create check: "This computer is running Hyper-V. Vi
rtualBox won't boot a 64bits VM when Hyper-V is activated. Either use Hyper-V as a driver, or disable the Hyper-V hypervisor. (To skip this check, use --virtualbox-no-vtx-check)". Retrying.
. FAIL E0913 12:36:16.093452   10688 start.go:339] Error starting the VM: Error creating the VM. Error with pre-create check: "This computer is running Hyper-V. VirtualBox won't boot a 64bits
VM when Hyper-V is activated. Either use Hyper-V as a driver, or disable the Hyper-V hypervisor. (To skip this check, use --virtualbox-no-vtx-check)". Retrying.
. FAIL E0913 12:36:18.447673   10688 start.go:339] Error starting the VM: Error creating the VM. Error with pre-create check: "This computer is running Hyper-V. VirtualBox won't boot a 64bits
VM when Hyper-V is activated. Either use Hyper-V as a driver, or disable the Hyper-V hypervisor. (To skip this check, use --virtualbox-no-vtx-check)". Retrying.
Error starting the VM: Error creating the VM. Error with pre-create check: "This computer is running Hyper-V. VirtualBox won't boot a 64bits VM when Hyper-V is activated. Either use Hyper-V as
 a driver, or disable the Hyper-V hypervisor. (To skip this check, use --virtualbox-no-vtx-check)"
Error creating the VM. Error with pre-create check: "This computer is running Hyper-V. VirtualBox won't boot a 64bits VM when Hyper-V is activated. Either use Hyper-V as a driver, or disable t
he Hyper-V hypervisor. (To skip this check, use --virtualbox-no-vtx-check)"
Error creating the VM. Error with pre-create check: "This computer is running Hyper-V. VirtualBox won't boot a 64bits VM when Hyper-V is activated. Either use Hyper-V as a driver, or disable t
he Hyper-V hypervisor. (To skip this check, use --virtualbox-no-vtx-check)"
```

### PR DESCRIPTION
**Pull request by anjannath**
Fixes #1376 
See [#1376/comment](https://github.com/minishift/minishift/issues/1376#issuecomment-363766107)

### CODE REVIEW

File: cmd/minishift/cmd/start_preflight.go [hunk 1]
```diff
@@ -357,14 +357,22 @@ func checkHypervDriverSwitch() bool {
 
 // checkHypervDriverInstalled returns true if Hyper-V driver is installed
 func checkHypervDriverInstalled() bool {
-	posh := powershell.New()
+	// check if hyper-v is installed
```
Comment (gbraad) [inline cmd/minishift/cmd/start_preflight.go]: // Check if Hyper-V's Virtual Machine Management Service is installed

**Comment by gbraad** [comment]
Can you describe which scenario is covered with the pre-flight test...

I have Hyper-V installed, but try to run VirtualBox? and what do we check... ?

**Comment by anjannath** [comment]
We check if hypervisor is present by using `(Get-Wmiobject Win32_ComputerSystem).HypervisorPresent`, if this turns out to be true, we exit with an error message saying something like `VirtualBox can not run when Hyper-V is installed. Please use Hyper-V or disable it to use VirtualBox`

**Comment by gbraad** [comment]
`Disable` is not the correct approach/comment as we need to deativate the feature (which means uninstall).

**Comment by anjannath** [comment]
>  `Disable` is not the correct approach/comment as they need to deativate the feature (which means uninstall).

how about this line?  _VirtualBox can not run when Hyper-V is installed. Please use Hyper-V or uninstall Hyper-V to use VirtualBox_

File: cmd/minishift/cmd/start_preflight.go [hunk 2]
```diff
@@ -357,13 +357,22 @@ func checkHypervDriverSwitch() bool {
 
 // checkHypervDriverInstalled returns true if Hyper-V driver is installed
 func checkHypervDriverInstalled() bool {
+	// Check if Hyper-V's Virtual Machine Management Service is installed
```
Comment (budhrg) [inline cmd/minishift/cmd/start_preflight.go]: `// check` Let's start comment with small letter.

**Comment by budhrg** [comment]
@anjannath , In following both occasions, I am able to start Minishift.
- Second scenario, only management tools are turned on
- Third scenario, only Hyper-V platform is turned on  

Reference comment https://github.com/minishift/minishift/issues/1376#issuecomment-363766107

File: cmd/minishift/cmd/start_preflight.go (hunk 2 above)
Comment (gbraad) [reply cmd/minishift/cmd/start_preflight.go]: No!

code comments start with lowercase, as it follows the casing. But textual comments should follow grammtic rules, so a sentence starts with a uppercase.

Comment (budhrg) [reply cmd/minishift/cmd/start_preflight.go]: sure

**Comment by gbraad** [comment]
@budhrg I am not following your assesment. Can you write as *Steps*, *Expected* and *Actual* for this?

**Comment by budhrg** [comment]
@anjannath , please check comment  https://github.com/minishift/minishift/pull/2009#pullrequestreview-96859995 and https://github.com/minishift/minishift/pull/2009#issuecomment-365943679

**Comment by LalatenduMohanty** [comment]
The code looks fine to me. @budhrg Feel free to merge it as my windows box is not ready yet to test this.

**Pull request merged**